File: extension/src/types/reply.ts
/**
 * Reply type definitions.
 */
export type Tone =
  | 'flirty'
  | 'confident'
  | 'funny'
  | 'casual'
  | 'formal'
  | 'witty'
  | 'sarcastic'
  | 'romantic'
  | 'friendly';

export interface SingleReply {
  index: number;
  text: string;
  tone: string;
}

export interface GenerateReplyResponse {
  replies: SingleReply[];
  credits_used: number;
  credits_remaining: number;
  model: string;
  latency_ms: number;
}

export interface ReplyHistoryItem {
  id: string;
  platform: string;
  tone: string;
  contact_name: string | null;
  conversation_snippet: string;
  replies: { items: string[] } | null;
  selected_reply_index: number | null;
  model_used: string;
  created_at: string;
}

export const TONE_CONFIG: Record<Tone, { label: string; emoji: string; color: string }> = {
  flirty: { label: 'Flirty', emoji: '😏', color: '#FF6B9D' },
  confident: { label: 'Confident', emoji: '💪', color: '#4A90D9' },
  funny: { label: 'Funny', emoji: '😂', color: '#FFB347' },
  casual: { label: 'Casual', emoji: '😎', color: '#77DD77' },
  formal: { label: 'Formal', emoji: '🎩', color: '#9B9B9B' },
  witty: { label: 'Witty', emoji: '🧠', color: '#BB77FF' },
  sarcastic: { label: 'Sarcastic', emoji: '😒', color: '#FF7744' },
  romantic: { label: 'Romantic', emoji: '❤️', color: '#FF4D6A' },
  friendly: { label: 'Friendly', emoji: '🤗', color: '#4ECDC4' },
};

File: backend/app/schemas/reply.py
"""
Pydantic schemas for reply generation endpoints.
"""

from __future__ import annotations

import uuid
from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, Field

from app.schemas.conversation import ConversationPayload


class Tone(str, Enum):
    FLIRTY = "flirty"
    CONFIDENT = "confident"
    FUNNY = "funny"
    CASUAL = "casual"
    FORMAL = "formal"
    WITTY = "witty"
    SARCASTIC = "sarcastic"
    ROMANTIC = "romantic"
    FRIENDLY = "friendly"


class GenerateReplyRequest(BaseModel):
    """Request body for reply generation."""
    conversation: ConversationPayload
    tone: Tone = Tone.CONFIDENT
    num_replies: int = Field(default=3, ge=1, le=5)
    max_length: Optional[int] = Field(default=None, ge=10, le=500)


class SingleReply(BaseModel):
    """One generated reply suggestion."""
    index: int
    text: str
    tone: str


class GenerateReplyResponse(BaseModel):
    """Response containing multiple reply suggestions."""
    replies: List[SingleReply]
    credits_used: int
    credits_remaining: int
    model: str
    latency_ms: float


class ReplyHistoryItem(BaseModel):
    """A past reply generation stored in history."""
    id: uuid.UUID
    platform: str
    tone: str
    contact_name: Optional[str]
    conversation_snippet: str
    replies: Optional[list]
    selected_reply_index: Optional[int]
    model_used: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ReplyHistoryListResponse(BaseModel):
    """Paginated list of reply history items."""
    items: List[ReplyHistoryItem]
    total: int
    page: int
    page_size: int
